db/machine.go

```diff
@@ -97,9 +338,84 @@ func (d DB) GetMachines() ([]Machine, error) {
 	return machines, rows.Err()
 }
 
+//FILTER LIST OF MACHINE: BY RSERVATION DATES
+func(d DB) Filter_M_By_Dates(from time.Time, to time.Time) ([]Machine, error){
+
+         rows, err := d.sql.Query(`
+		SELECT m.id, name, arch, microarch, cores, memory
+		FROM Machines m
+		WHERE m.id NOT IN (SELECT r.machine FROM Reservations r
+		WHERE (? BETWEEN r.start AND r.end)
+		OR (? BETWEEN r.start AND r.end)
+		AND  ( r.start BETWEEN ? AND ?) OR (r.end BETWEEN ? AND ?)
+	
+	);	
+	`, from, to, from, to, from, to)
+	if err != nil {
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	fm := []Machine{}
+	for rows.Next() {
+
+		f := Machine{db: &d}
+		err = rows.Scan(
+			&f.Id, &f.Name, &f.Architecture, &f.Microarchitecture,
+			&f.Cores, &f.MemoryGB)
+		if err != nil {
+			return nil, err
+		}
+
+		fm = append(fm, f)
+	}
+
+	return fm, rows.Err()
+
+
+}
+
+//FILTER LIST OF MACHINES: GET AVAILABLE  MACHINES 
+func (d DB) AvailableMachines()([]Machine, error){
```

Comment:
This method is redundant: we can just call the version with parameters and pass it `from = currentTime` and `to = currentTime`.